packages/core/strapi/lib/core-api/controller/index.js

```diff
@@ -1,20 +1,34 @@
 'use strict';
 
-const { sanitizeEntity, contentTypes } = require('@strapi/utils');
+const { getOr } = require('lodash/fp');
+
+const { contentTypes, sanitize } = require('@strapi/utils');
 
 const { transformResponse } = require('./transform');
 const createSingleTypeController = require('./single-type');
 const createCollectionTypeController = require('./collection-type');
 
+const getAuthFromKoaContext = getOr({}, 'state.auth');
+
 module.exports = ({ service, model }) => {
   const ctx = {
     model,
     service,
+
     transformResponse(data, meta) {
       return transformResponse(data, meta, { contentType: model });
     },
-    sanitize(data) {
-      return sanitizeEntity(data, { model: strapi.getModel(model.uid) });
+
+    sanitizeOutput(data, ctx) {
+      const auth = getAuthFromKoaContext(ctx);
+
+      return sanitize.contentAPI.output(data, strapi.getModel(model.uid), { auth });
```

Comment:
why would this need to by async ?

Reply:
Because when we're sanitizing the relations of an entity, we're using `strapi.auth.verify` to check the permissions.